### PR TITLE
Fix Playwright E2E tests - SideEffects

### PR DESCRIPTION
**Pull request by wavvves**
## What

Fix broken Playwright tests due to new Order Confirmation page

## Testing Instructions

1. Verify tests pass

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [x] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> n/a

### CODE REVIEW

**Review by gigitux — COMMENTED** [review]
The code LGTM, but before merging, ensure that E2E tests pass 🙏

File: src/BlockTypes/OrderConfirmation/Totals.php
Comment (opr) [inline src/BlockTypes/OrderConfirmation/Totals.php]: Could you explain the context of this change? Why remove this filter, what was it there for originally and why is it no longer needed?

Comment (wavvves) [reply src/BlockTypes/OrderConfirmation/Totals.php]: Tests were failing because on the new Order Confirmation page, it wasn't displaying the "via xxx" text next to the shipping total. I thought about removing the assert, but that wouldn't be respecting the test. For the sake of fixing tests I prefer to merge this into trunk, and finding out later with the team the reason to remove that span.

Comment (wavvves) [reply src/BlockTypes/OrderConfirmation/Totals.php]: I went the other way around, removed text was by design requirement. Just waiting for tests to complete again

cc @mikejolley

**Review by opr — APPROVED** [review]
Looks OK - happy to merge once tests pass

File: src/BlockTypes/OrderConfirmation/Totals.php
Comment (ralucaStan) [reply src/BlockTypes/OrderConfirmation/Totals.php]: @wavvves can you link to that design requirement or add a pic?

A reminder that we are not skipping PW tests, and we are not taking shortcuts when dealing with failing E2E tests. We should check they don't catch regressions and address the bug fix accordingly. If the test is not reflecting the respected result, tests should be updated.

**Pull request merged**
